### PR TITLE
Add description of the native "Add X/Y fields to layer" algorithm

### PR DESCRIPTION
**Pull request by havatv**
Fixes: #3577

Goal:

Ticket(s): #

- [ ] Backport to LTR documentation is required

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 1]
```diff
@@ -158,6 +158,69 @@ Outputs
...
   Default: *Skip Output*
 
+
+.. _qgisaddxyfieldstolayer:
+
+Add X/Y fields to layer
+-----------------------
+Adds X and Y (or latitude/longitude) fields to a point layer.
+The X/Y fields can be calculated in a different CRS to the layer
+(e.g. creating latitude/longitude fields for a layer in a project CRS).
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Should this be 
```suggestion
(e.g. creating latitude/longitude fields for a layer in a projected CRS).
```

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 2]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+   :widths: 20 20 20 40
+   :stub-columns: 0
+
+   *  - Name
+      - Identifier
+      - Type
+      - Description
+   *  - **Input layer**
+      - INPUT
+      - [vector:any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]:
```suggestion
      - [vector: point]
```
Based on description above

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 3]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+      - The input layer.
+   *  - **Coordinate system**
+      - CRS
+      - [string]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]:
```suggestion
      - [crs]
```
Please give a look to the processing section in guidelines to figure out the tags to use.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 4]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+        
+        Default: "EPSG:4326"
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Are you sure that's the default? Not the layer CRS instead? Would make more sense to me. Worth a test/check imho.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 5]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+      - Join layer fields to copy (if empty, all fields will be
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: ???

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 6]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+        copied).
+   *  - **Field prefix**
+        
+        Optional
+      - PREFIX
+      - [vector:any]
+      - The join layer.
+   *  - **Added fields**
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Is it not in the output section?

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 7]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+      - OUTPUT
+      - [vector:point]
+      - Choice between ``Create Temporary Layer``, ``Save to File``,
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: I don't think these information should be part of the alg description but instead at a place we describe widgets or parameter types

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 8]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+        
+        Default: "EPSG:4326"
+      - Join layer fields to copy (if empty, all fields will be
+        copied).
+   *  - **Field prefix**
+        
+        Optional
+      - PREFIX
+      - [vector:any]
+      - The join layer.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Euh... no.
You have a speech about prefix in another PR that may be inspiring

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 9]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+      - [string]
+        
+        Default: "EPSG:4326"
+      - Join layer fields to copy (if empty, all fields will be
+        copied).
+   *  - **Field prefix**
+        
+        Optional
+      - PREFIX
+      - [vector:any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]: This should be a string, right?

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst [hunk 10]
```diff
@@ -158,6 +158,69 @@ Outputs
...
+
+   *  - Name
+      - Identifier
+      - Type
+      - Description
+   *  - **Added fields**
+      - OUTPUT
+      - [vector:point]
+      - The output layer - identical to the input layer but with two
+        new double fields, ``y`` and ``y``.
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectortable.rst]:
```suggestion
        new double fields, ``x`` and ``y``.
```

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: I think this is OK. I must have taken it from the code.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 6 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: It is also a parameter (it has to be a parameter if the user shall be able to specify the sink).

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 7 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: I agree, we can remove this when we have this in place.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 9 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Yes.  Will fix.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Definitely an error.  Will try to fix.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 9 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Should be fixed now.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: Should be fixed now.

File: source/docs/user_manual/processing_algs/qgis/vectortable.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/processing_algs/qgis/vectortable.rst]: According to the code (and I have also tested) it is in fact EPSG:4326.

**Comment by havatv** [comment]
Thanks, @DelazJ!

**Pull request merged**